server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -51,78 +68,90 @@ public class RemoteStoreRestoreService {
 
     private final AllocationService allocationService;
 
-    public RemoteStoreRestoreService(ClusterService clusterService, AllocationService allocationService) {
+    private final MetadataCreateIndexService createIndexService;
+
+    private final MetadataIndexUpgradeService metadataIndexUpgradeService;
+
+    private final ShardLimitValidator shardLimitValidator;
+
+    private final ClusterSettings clusterSettings;
+
+    public RemoteStoreRestoreService(
+        ClusterService clusterService,
+        AllocationService allocationService,
+        MetadataCreateIndexService createIndexService,
+        MetadataIndexUpgradeService metadataIndexUpgradeService,
+        ClusterSettings clusterSettings,
+        ShardLimitValidator shardLimitValidator
+    ) {
         this.clusterService = clusterService;
         this.allocationService = allocationService;
+        this.createIndexService = createIndexService;
+        this.metadataIndexUpgradeService = metadataIndexUpgradeService;
+        this.clusterSettings = clusterService.getClusterSettings();
+        this.shardLimitValidator = shardLimitValidator;
```

Comment:
Where are we using `clusterSettings` and `shardLimitValidator`?

Reply:
We are not, was planning to use them earlier, but will remove now

Reply:
using shardLimitValidator now, removed clusterSettings